app/uk/gov/hmrc/helptosavefrontend/connectors/NSIConnector.scala

```diff
@@ -130,6 +131,18 @@ class NSIConnectorImpl @Inject() (conf: Configuration, metrics: Metrics) extends
       }
   }
 
+  override def healthCheck(userInfo: NSIUserInfo)(implicit hc: HeaderCarrier, ex: ExecutionContext): Result[Unit] = EitherT[Future, String, Unit]{
+    httpProxy.put(nsiHealthCheckUrl, userInfo, Map(nsiAuthHeaderKey → nsiBasicAuth))(nsiUserInfoFormat, hc.copy(authorization = None), ex)
+      .map[Either[String, Unit]] { response ⇒
+        response.status match {
+          case Status.OK ⇒ Right(())
+          case other     ⇒ Left(s"Received unexpected status $other from NS&I while trying to update email. Body was ${response.body}")
+        }
+      }.recover {
+        case e ⇒ Left(s"Encountered error while trying to create account: ${e.getMessage}")
```

Comment:
Is this text correct? Maybe...
HealthCheck encountered an error while trying to communicate with NS&I: ${e.getMessage}